bin/lint-migrations-file/src/lint_migrations_file.clj

```diff
@@ -61,9 +59,10 @@
 
       ;; a modifyDataType change; see if it changes a column to target-type
       (:modifyDataType x)
-      (if (= target-types (str/lower-case (or (get-in x [:modifyDataType :newDataType]) "")))
-        (do (swap! found-cols conj x)
-            x)
+      (if (contains? target-types (str/lower-case (or (get-in x [:modifyDataType :newDataType]) "")))
+        (do
+         (swap! found-cols conj x)
+         x)
```

Comment:
small nit: If I'm reading this right, I think you can reorganize this expression a bit to avoid repeating the result value.

```clojure
(do 
  (when (contains? target-types (str/lower-case (or (get-in x [:modifyDataType :newDataType]) "")))
    (swap! found-cols conj x))
  x
)
```

I'm also not sure if you're intentionally guarding against falsey`newDataType` values, but perhaps the `or` could be replaced by passing a default value to `get-in`?

Reply:
Actually, it looks like *every* case in the cond is returning x? Perhaps the `do` can move around that, so the cond *only* does side-effects.

Reply:
I there are two places where this `target-types` check is done, so it could be nice to break out a helper function to encapsulate the lowercase and empty string logic?

```clojure
(defn- target-type? [target-types raw-type]
  (contains? target-types (str/lower-case (or raw-type ""))))
```

Reply:
> Actually, it looks like every case in the cond is returning x? Perhaps the do can move around that, so the cond only does side-effects.

Oh, maybe even like that:

```
(when (map? x)
  (cond ...))
x
```

Reply:
thanks for the recommendations, I incorporated them in d441a1ebc2